src/claimable_balances/single_claimable_balance_request.rs

```diff
@@ -1,30 +1,43 @@
 use crate::models::*;
 
+
+#[derive(Default, Clone)]
+pub struct ClaimableBalanceId(String);
+#[derive(Default, Clone)]
+pub struct NoClaimableBalanceId;
+
 /// SingleClaimableBalanceRequest is the struct that implements the type for the /claimable_balances endpoint to get a single claimable balance
-/// [More Details](https://laboratory.stellar.org/#explorer?resource=claimable_balances&endpoint=single&network=test "Single Claimable Balance")
-#[derive(Debug)]
-pub struct SingleClaimableBalanceRequest {
-    /// Claimable Balance ID
-    /// [Stellar Documentation](https://developers.stellar.org/api/resources/claimablebalances/single/ "Claimable Balance ID")
-    claimable_balance_id: Option<String>,
+/// [More Details](https://developers.stellar.org/api/horizon/resources/retrieve-a-claimable-balance) "Single Claimable Balance")
+#[derive(Default)]
+pub struct SingleClaimableBalanceRequest<I> {
+    claimable_balance_id: I,
+}
+
+impl SingleClaimableBalanceRequest<NoClaimableBalanceId> {
+    pub fn new() -> Self {
+        SingleClaimableBalanceRequest::default()
+    }
 }
 
-impl Request for SingleClaimableBalanceRequest {
-    fn new() -> Self {
-        SingleClaimableBalanceRequest {
-            claimable_balance_id: None,
+impl<I> SingleClaimableBalanceRequest<I> {
+    pub fn set_claimable_balance_id(
+        self,
+        claimable_balance_id: String,
+    ) -> SingleClaimableBalanceRequest<ClaimableBalanceId> {
+        SingleClaimableBalanceRequest { 
+            claimable_balance_id: ClaimableBalanceId(claimable_balance_id) 
```

Comment:
What would be really awesome is that we could implement the `set_claimable_balance_id` on the implementation of `SingleClaimableBalanceRequest<NoClaimableBalanceId>`. Once the balance is set, there is way to set it again, as this function if no longer implemented for `SingleClaimableBalanceRequest<ClaimableBalanceId>`

This trick might apply to other request builders too.

Reply:
Made non-optional parameters immutable!